extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -558,25 +558,34 @@ export class AmpA4A extends AMP.BaseElement {
     // So long as any one of these signing services can verify the
     // signature, then the creative is valid AMP.
     const keyInfoSetPromises = this.win.ampA4aValidationKeys;
+    // Track if verification found as it will ensure that promises yet to
+    // resolve will "cancel" as soon as possible saving unnecessary resource
+    // allocation.
+    let verificationFound = false;
+    const alreadyVerified = () => {
+      return verificationFound ?
+          Promise.reject('verification already found') : null;
+    };
     return some(keyInfoSetPromises.map(keyInfoSetPromise => {
       // Resolve Promise into Array of Promises of signing keys.
       return keyInfoSetPromise.then(keyInfoSet => {
```

Comment:
Would it be worth having an `alreadyVerified` check here as well?  I still am not sure I'm tracking the logic of this method fully, but I think you can be in the case where a previous `keyInfoSetPromise` has already verified the sig.  Is that right?  If so, you could skip out entirely here.